app/ts/utils/ethereumWallet.ts

```diff
@@ -0,0 +1,33 @@
+import 'viem/window'
+import { AccountAddress } from '../types/types.js'
+import { createPublicClient, createWalletClient, custom, http, publicActions } from 'viem'
+import { mainnet } from 'viem/chains'
+
+export const requestAccounts = async () => {
+	if (window.ethereum === undefined) throw new Error('no window.ethereum injected')
+	const reply = await window.ethereum.request({ method: 'eth_requestAccounts', params: undefined })
+	return reply[0]
+}
+
+export const getAccounts = async () => {
+	if (window.ethereum === undefined) throw new Error('no window.ethereum injected')
+	const reply = await window.ethereum.request({ method: 'eth_accounts', params: undefined })
+	return reply[0]
+}
+
+export const createReadClient = (accountAddress: AccountAddress | undefined) => {
+	if (window.ethereum === undefined || accountAddress === undefined) {
+		return createPublicClient({ chain: mainnet, transport: http('https://ethereum.dark.florist', { batch: { wait: 100 } }) })
+	}
+	return createWalletClient({ chain: mainnet, transport: custom(window.ethereum) }).extend(publicActions)
+}
+
+export const createWriteClient = (accountAddress: AccountAddress) => {
+	if (window.ethereum === undefined) throw new Error('no window.ethereum injected')
+	if (accountAddress === undefined) throw new Error('no accountAddress!')
+	return createWalletClient({ account: accountAddress, chain: mainnet, transport: custom(window.ethereum) }).extend(publicActions)
+}
+
+export const getChainId = async (accountAddress: AccountAddress) => {
+	return await createWriteClient(accountAddress).getChainId()
```

Comment:
This could be a read client couldn't it? Also just to consider it feels like it might be better to take a client as its argument instead of an address, but prob not that big a deal to not.

Reply:
agree. It has been done this way because of a viem bug that is now fixed as we bumped the version